config/gradle/publish.gradle

```diff
@@ -17,16 +17,45 @@ publishing {
+                        println "Changing PUBLISH repoKey set via Gradle property to $publishRepo"
                     } else {
-                        url = 'http://artifactory.terasology.org/artifactory/terasology-snapshot-local'
-                        println "PUBLISH repoKey is terasology-snapshot-local (default value)"
+                        // In this case we are going to fletch the publish repo together from a few things
+                        // First if we have an override from the environment to use a different target publish org
+                        String deducedPublishRepo = System.getenv()["PUBLISH_ORG"]
+                        if (deducedPublishRepo == null || deducedPublishRepo == "") {
+                            // If not then default
+                            deducedPublishRepo = "terasology"
+                        }
+                        String releaseRepoFragment = "-release-local"
+                        String snapshotRepoFragment = "-snapshot-local"
+
+                        // Secondly we're going to find out if we're doing a release or a snapshot - this gets a little more complicated
+                        String gitBranch = System.getenv()["BRANCH_NAME"]
+                        if (gitBranch != null && gitBranch != "" && gitBranch.equals("master")) {
+                            // Okay we're in an environment where a branch name is set to 'master' so we might be doing a release
+                            // This is the funny part. Modules aren't ready globally to accept master branch == release, so check a prop that defaults to bypass
+                            if (project.hasProperty("bypassModuleReleaseManagement")) {
+                                if (bypassModuleReleaseManagement == "true") {
+                                    println "Release management not enabled for " + project.name + ", using snapshot repo despite building 'master' branch"
+                                    deducedPublishRepo += snapshotRepoFragment
+                                } else {
+                                    println "Release management *is* enabled for " + project.name + ", using release repo since building 'master' branch"
+                                    deducedPublishRepo += releaseRepoFragment
+                                }
+                            } else {
+                                println "We're working on a 'master' branch with defaults so using the release publish repo"
+                                deducedPublishRepo += releaseRepoFragment
+                            }
+                        } else {
+                            // No master branch so we for sure are just doing snapshots
+                            println "We're not working with a branch name set that's 'master' so assuming we're publishing snapshots"
+                            deducedPublishRepo += "-snapshot-local"
+                        }
```

Comment:
The whole logic in here boils down to just this, doesn't it?

```groovy
if (isMaster(maybeGitBranch) && !shouldBypassModuleRelease()) {
	deducedPublishRepo += "-release-local"
} else {
	deducedPublishRepo += "-snapshot-local"
}
```

I think with the two additional functions this code is really readable, and the nesting is reduced.

```groovy
def isMaster(gitBranch) {
  return gitBranch != null && gitBranch != "" && gitBranch.equals("master");
}

def shouldBypassModuleRelease() {
  return project.hasProperty("bypassModuleReleaseManagement") && 
	bypassModuleReleaseManagement == "true";
}
```

Reply:
This would loose a couple of console lines, but overall the structure in the build file is more comprehensible imho.

Reply:
Thank you! I had even _more_ logic madness in there while trying to find a solution that would cover all the scenarios, this was actually an already boiled down version, but closing in on 2 am was not letting me see clearly in shrinking it further. This is way cleaner! Don't need the logging now that it tests out fine :-)